data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/ScrollWorker.java

```diff
@@ -63,40 +64,55 @@ public ScrollWorker(final SearchAccessor searchAccessor,
-            }
 
-            try {
-                processIndex(indexPartition.get());
-
-                sourceCoordinator.closePartition(
-                        indexPartition.get().getPartitionKey(),
-                        openSearchSourceConfiguration.getSchedulingParameterConfiguration().getRate(),
-                        openSearchSourceConfiguration.getSchedulingParameterConfiguration().getJobCount());
-            } catch (final PartitionUpdateException | PartitionNotFoundException | PartitionNotOwnedException e) {
-                LOG.warn("ScrollWorker received an exception from the source coordinator. There is a potential for duplicate data for index {}, giving up partition and getting next partition: {}", indexPartition.get().getPartitionKey(), e.getMessage());
-                sourceCoordinator.giveUpPartitions();
-            } catch (final SearchContextLimitException e) {
-                LOG.warn("Received SearchContextLimitExceeded exception for index {}. Giving up index and waiting {} seconds before retrying: {}",
-                        indexPartition.get().getPartitionKey(), BACKOFF_ON_SCROLL_LIMIT_REACHED.getSeconds(), e.getMessage());
-                sourceCoordinator.giveUpPartitions();
                 try {
-                    Thread.sleep(BACKOFF_ON_SCROLL_LIMIT_REACHED.toMillis());
+                    processIndex(indexPartition.get());
+
+                    sourceCoordinator.closePartition(
+                            indexPartition.get().getPartitionKey(),
+                            openSearchSourceConfiguration.getSchedulingParameterConfiguration().getRate(),
+                            openSearchSourceConfiguration.getSchedulingParameterConfiguration().getJobCount());
+                } catch (final PartitionUpdateException | PartitionNotFoundException | PartitionNotOwnedException e) {
+                    LOG.warn("ScrollWorker received an exception from the source coordinator. There is a potential for duplicate data for index {}, giving up partition and getting next partition: {}", indexPartition.get().getPartitionKey(), e.getMessage());
+                    sourceCoordinator.giveUpPartitions();
+                } catch (final SearchContextLimitException e) {
+                    LOG.warn("Received SearchContextLimitExceeded exception for index {}. Giving up index and waiting {} seconds before retrying: {}",
+                            indexPartition.get().getPartitionKey(), BACKOFF_ON_SCROLL_LIMIT_REACHED.getSeconds(), e.getMessage());
+                    sourceCoordinator.giveUpPartitions();
+                    try {
+                        Thread.sleep(BACKOFF_ON_SCROLL_LIMIT_REACHED.toMillis());
+                    } catch (final InterruptedException ex) {
+                        return;
+                    }
+                } catch (final IndexNotFoundException e){
+                    LOG.warn("{}, marking index as complete and continuing processing", e.getMessage());
```

Comment:
Should this be an error level?

Reply:
I wouldn't say so, because in many cases it may be intentional that the user would delete the index and not want it processed.